### PR TITLE
Feature/#6 reporting

### PR DESCRIPTION
**Pull request by kailanwyatt**
Fixes #17

### CODE REVIEW

File: includes/bp-media-function.php [hunk 1]
```diff
@@ -602,10 +602,9 @@ function bp_media_group_image_link( $id, $user_id ) {
 	echo bp_media_get_group_image_link( $id, $user_id );
 }
 
-	function bp_media_get_group_image_link( $id, $user_id ) {
-		return  bp_core_get_user_domain( $user_id ) . BP_MEDIA_SLUG . '/image/' . $id;
-	}
-
+function bp_media_get_group_image_link( $id, $user_id ) {
+	return  bp_core_get_user_domain( $user_id ) . BP_MEDIA_SLUG . '/image/' . $id;
```
Comment (tw2113) [inline includes/bp-media-function.php]: let's get rid of the extra space in front of the return. Only needs one.

File: includes/bp-media-function.php [hunk 2]
```diff
@@ -725,13 +720,15 @@ function bp_media_display_attachment_image() {
...
 
 	$subposts = get_children(array(
-	    'post_parent' => $id,
-	    'post_type'   => 'any',
-	    'numberposts' => -1,
-	    'post_status' => 'any'
+		'post_parent' => $id,
+		'post_type'   => 'any',
+		'numberposts' => -1,
+		'post_status' => 'any'
```
Comment (tw2113) [inline includes/bp-media-function.php]: nitpicky trailing comma missing

File: includes/bp-media-function.php (hunk 1 above)
Comment (tw2113) [reply includes/bp-media-function.php]: Also a FYI regarding the indentation of some of these function declarations. It is a holdover of sorts from how BuddyPress core groups together their "echo" and "return" functions. They indent the declaration for the return versions.

File: includes/bp-media-function.php [hunk 3]
```diff
@@ -859,45 +856,45 @@ function bp_is_friend_boolean() {
...
+function bp_media_get_pagination_count( $query ) {
 
-		if( bp_is_directory() && !bp_current_action() ) {
-			$action = __('album', 'bp-media');
-		}
+	$action = ( 'album' !== bp_current_action() && bp_is_user() ) ? __('album', 'bp-media') : __('image', 'bp-media') ;
 
-		$paged = ( isset($_GET['mpage']) ) ? $_GET['mpage'] : 1;
-		$posts_per_page = $query->query['posts_per_page'];
+	if( bp_is_directory() && !bp_current_action() ) {
```
Comment (tw2113) [inline includes/bp-media-function.php]: missing space after this if keyword

File: includes/bp-media-function.php [hunk 4]
```diff
@@ -859,45 +856,45 @@ function bp_is_friend_boolean() {
...
+		$action = __('album', 'bp-media');
+	}
 
-		$start_num = intval( ( $paged - 1 ) * $posts_per_page ) + 1;
-		$from_num  = bp_core_number_format( $start_num );
-		$to_num    = bp_core_number_format( ( $start_num + ( $posts_per_page - 1 ) > $query->found_posts ) ? $query->found_posts : $start_num + ( $posts_per_page - 1 ) );
-		$total     = bp_core_number_format( $query->found_posts );
+	$paged = ( isset($_GET['mpage']) ) ? $_GET['mpage'] : 1;
```
Comment (tw2113) [inline includes/bp-media-function.php]: let's absint() the $_GET param if it's available.

File: includes/bp-media-function.php [hunk 5]
```diff
@@ -908,26 +905,62 @@ function bp_media_pagination_links( $query ) {
 	echo bp_media_get_pagination_links( $query );
 }
 
-	function bp_media_get_pagination_links( $query ) {
+function bp_media_get_pagination_links( $query ) {
 
-			$paged = ( isset($_GET['mpage']) ) ? $_GET['mpage'] : 1;
+	$paged = ( isset( $_GET['mpage'] ) ) ? $_GET['mpage'] : 1;
```
Comment (tw2113) [inline includes/bp-media-function.php]: another $_GET param to absint()

File: includes/bp-media-function.php [hunk 6]
```diff
@@ -908,26 +905,62 @@ function bp_media_pagination_links( $query ) {
...
-			if ( defined( 'DOING_AJAX' ) && true === (bool) DOING_AJAX ) {
-				$base = remove_query_arg( 's', wp_get_referer() );
-			} else {
-				$base = '';
-			}
+	if ( defined( 'DOING_AJAX' ) && true === (bool) DOING_AJAX ) {
+		$base = remove_query_arg( 's', wp_get_referer() );
+	} else {
+		$base = '';
+	}
```
Comment (tw2113) [inline includes/bp-media-function.php]: let's set $base to empty string above this if statement, and remove the need for an else.

Comment (kailanwyatt) [reply includes/bp-media-function.php]: I'll go ahead and do PHPCS clean on the entire plugin. Will take me a bit but should solve some issues and do some validation updates.

File: includes/templates/media/single/image.php [hunk 7]
```diff
@@ -6,16 +6,19 @@
 if( $post ) : ?>
 
 <div class="media-pop-wrapper">
+	<?php do_action( 'bp_media_before_photo_view' ); ?>
 	<div class="photo-column">
 		<div class="image-options">
 			<a href="<?php bp_media_album_back_url( $post->post_parent ) ;?>" class="left"><?php _e( 'back to album', 'bp_media' ) ; ?></a>
+			<?php do_action( 'bp_media_photo_options' ); ?>
```
Comment (tw2113) [inline includes/templates/media/single/image.php]: Lets get these two action hooks phpdocblocked

**Review by tw2113 — APPROVED** [review]
Let's do it to it.

**Pull request merged**